### ISSUE TITLE
Provide more latency data in results

### ISSUE THREAD

**Issue by zhouyuan** [RedisLabs/memtier_benchmark#68]
Hi memtier_benchmark developers, 

Currently memtier_benchmark can report average latency in the results, which is usually enough for simple benchmarks. it also provides the option to print the detail latency distribution. The data is aggregated through different latency level. However for storage/DB performance, people will also care about the tail latency(99%, 99.9% or 99.99%,  [example](https://github.com/axboe/fio/blob/307f2246d65d9dbea7e5f56cd17734cb68c2817f/HOWTO#L3528) from fio).  Is it possible to provide these tail latency data also? 

```
ALL STATS
========================================================================
Type        Ops/sec     Hits/sec   Misses/sec      Latency       KB/sec
------------------------------------------------------------------------
Sets       21536.09          ---          ---      0.23100     22561.95
Gets           0.00         0.00         0.00      0.00000         0.00
Waits          0.00          ---          ---      0.00000          ---
Totals     21536.09         0.00         0.00      0.23100     22561.95
```
thanks, -yuan

**Comment by yaacovhazan-Redislabs** [comment]
Hi @zhouyuan, we think that this is a good idea, and can really help the users to understand the benchmark results.
We will investigate how to best implement your suggestion for practical use.

**Comment by zhouyuan** [comment]
Thanks a lot! @yaacovhazan-Redislabs 


FWIW I did some naive hack here to extract the 99% latency data, not that accurate but can help me to do more analysis with the results

```
diff --git a/run_stats.cpp b/run_stats.cpp
index 3b89a14..de82c6f 100644
--- a/run_stats.cpp
+++ b/run_stats.cpp
@@ -38,6 +38,7 @@

 #define MIN(a,b) (((a)<(b))?(a):(b))
 #define MAX(a,b) (((a)>(b))?(a):(b))
+#define TAIL_LAT 99.00

 void output_table::add_column(table_column& col) {
     assert(columns.empty() || columns[0].elements.size() == col.elements.size());
@@ -867,20 +868,34 @@ void run_stats::print(FILE *out, bool histogram,
         } else {
             // SETs
             // ----
+           double prev = 0;
+           double curr = 0;
             if (jsonhandler != NULL){ jsonhandler->open_nesting("SET",NESTED_ARRAY);}
             for( latency_map_itr_const it = m_set_latency_map.begin() ; it != m_set_latency_map.end() ; it++) {
                 total_count += it->second;
+               curr = (double) total_count / m_totals.m_set_cmd.m_ops * 100;
+               if (prev < TAIL_LAT && curr > TAIL_LAT) {
+                  histogram_print(out, jsonhandler, "SET",it->first, TAIL_LAT);
+                }
                 histogram_print(out, jsonhandler, "SET",it->first,(double) total_count / m_totals.m_set_cmd.m_ops * 100);
+               prev = curr;
             }
             if (jsonhandler != NULL){ jsonhandler->close_nesting();}
             fprintf(out, "---\n");
             // GETs
             // ----
             total_count = 0;
+           prev = 0;
+           curr = 0;
             if (jsonhandler != NULL){ jsonhandler->open_nesting("GET",NESTED_ARRAY);}
             for( latency_map_itr_const it = m_get_latency_map.begin() ; it != m_get_latency_map.end() ; it++) {
                 total_count += it->second;
+               curr = (double) total_count / m_totals.m_get_cmd.m_ops * 100;
+               if (prev < TAIL_LAT && curr > TAIL_LAT) {
+                  histogram_print(out, jsonhandler, "GET",it->first, TAIL_LAT);
+                }
                 histogram_print(out, jsonhandler, "GET",it->first,(double) total_count / m_totals.m_get_cmd.m_ops * 100);
+               prev = curr;
             }
             if (jsonhandler != NULL){ jsonhandler->close_nesting();}
             fprintf(out, "---\n");
 
```

**Comment by filipecosta90** [comment]
Hi there @zhouyuan , following the just merged #100 you can now control the exposed overall latency metrics with the `--print-percentiles` flag. By default, and apart from the average latency, it will print the p50, p99 and p999, as follows:

```
memtier_benchmark  --hide-histogram --key-pattern=P:P --ratio 1:1
Writing results to stdout
[RUN #1] Preparing benchmark client...
[RUN #1] Launching threads now...
[RUN #1 100%,  13 secs]  0 threads:     2000000 ops,  120292 (avg:  143312) ops/sec, 8.76MB/sec (avg: 10.44MB/sec),  1.66 (avg:  1.39) msec latency

4         Threads
50        Connections per thread
10000     Requests per client


ALL STATS
============================================================================================================================
Type         Ops/sec     Hits/sec   Misses/sec    Avg. Latency     p50 Latency     p99 Latency   p99.9 Latency       KB/sec 
----------------------------------------------------------------------------------------------------------------------------
Sets        71598.09          ---          ---         1.39530         1.33500         3.31100         6.29500      5515.56 
Gets        71598.09     71598.09         0.00         1.39364         1.33500         3.30300         6.27100      5165.96 
Waits           0.00          ---          ---             ---             ---             ---             ---          --- 
Totals     143196.17     71598.09         0.00         1.39447         1.33500         3.30300         6.28700     10681.52
```
